### ISSUE TITLE
Health check from other service configured with retry policy to config server can hang.

### ISSUE THREAD

**Issue by wblancqu** [spring-cloud/spring-cloud-config#510]
We encountered some strange behavior when we have a service, which is configured to contact the config-service with a retry policy (for instance try for 30 minutes) with regards to that service health check page.

Looks like the same retry policy is used for updating the health information of the config service in that service. However, when the config service is down, this check is retrying and retrying. At that time, the /health page of the service hangs and doesn't show.

We use external automated monitoring on /health to identify the health status. In the above case, it marks the service as down, while it isn't.

I'd guess the health check towards the config server should not take into account the configured retry policy and just show DOWN instantly when it is not there.

Kind regards,
Wouter

**Comment by ryanjbaxter** [comment]
It should eventually give up though.  Do you have `spring.cloud.config.failFast=true` set in your config client?

**Comment by wblancqu** [comment]
These are our settings:

spring:
  cloud:
    config:
      failFast: true
      retry:
        multiplier: 1.1
        maxAttempts: 100
        maxInterval: 30000

I guess we could indeed fiddle around with those to let the health check page hang less time, but I have the feeling that would only be a work-around.

**Comment by ryanjbaxter** [comment]
I guess the question is how fast do u want to give up trying to connect to the config server if it is down.  I don't think you want it to keep retrying too long.

**Comment by wblancqu** [comment]
You're right about that. But the issue is that, even if it would only try for 2 minutes, that the /health page of the service is unavailable.

**Comment by ryanjbaxter** [comment]
I see your point as well.  @dsyer or @spencergibb does it make sense to try and make the health endpoint available while we continue to try to connect to the config server.  I havent looked at the code, but I can imagine we must be able to return a down/unknown status from the interface while we continue to try and make a connection.

**Comment by spencergibb** [comment]
In this case, I would disable the config server health check. The retry is annotation based. I'm unsure how to selectively turn off retry. #260 is related.

**Comment by dsyer** [comment]
Since this is a problem, in principle, for all health indicators, maybe we should look for a solution in spring boot? OTOH it seems like a really bad thing that we made it worse with the retry interceptor. There are ways to solve the problem in Spring Cloud as well. For instance, our health indicator could be aware of the problem and try to extract the target from the proxy and call it directly. Or it could create a new instance of the property source locator. That doesn't seem like a bad idea?

**Comment by ryanjbaxter** [comment]
I think if the problem can be fixed in spring boot than I think that sounds like the best solution,
